charty/src/commonMain/kotlin/com/himanshoe/charty/circle/CircleChart.kt

```diff
@@ -0,0 +1,131 @@
+package com.himanshoe.charty.circle
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.util.fastForEachIndexed
+import com.himanshoe.charty.circle.model.CircleData
+import kotlin.math.PI
+import kotlin.math.cos
+import kotlin.math.min
+import kotlin.math.sin
+
+/**
+ * A composable function that displays a circle chart based on the provided data.
+ *
+ * @param data A lambda function that returns a list of CircleData objects representing the data to be displayed.
+ * @param modifier A Modifier to be applied to the Canvas.
+ * @param onCircleClick A lambda function to handle click events on the circle. It receives the corresponding `CircleData` as parameters.
+ */
+@Composable
+fun CircleChart(
+    data: () -> List<CircleData>,
+    modifier: Modifier = Modifier,
+    showEndIndicator: Boolean = true,
+    onCircleClick: (CircleData) -> Unit = {}
+) {
+    CircleChartContent(
+        modifier = modifier,
+        onCircleClick = onCircleClick,
+        data = data,
+        showEndIndicator = showEndIndicator,
+    )
+}
+
+@Composable
+private fun CircleChartContent(
+    data: () -> List<CircleData>,
+    modifier: Modifier = Modifier,
+    showEndIndicator: Boolean = true,
+    onCircleClick: (CircleData) -> Unit = {}
+) {
+    var clickedCircleIndex by remember { mutableStateOf(-1) }
+    val circleData = remember(data) { data() }
```

Comment:
This can cause issues. If the data changes, `circleData` won't change because it's referencing the "lambda instance" and not the "data instance". So this is may be a bug! You can check once.

it should be simply like:

```
val circleData = data()
```